components/malloc_size_of/lib.rs

```diff
@@ -825,6 +826,183 @@ impl<Static: string_cache::StaticAtomSet> MallocSizeOf for string_cache::Atom<St
     }
 }
 
+impl MallocSizeOf for usvg::Tree {
+    fn size_of(&self, ops: &mut MallocSizeOfOps) -> usize {
+        let root = self.root();
+        let linear_gradients = self.linear_gradients();
+        let radial_gradients = self.radial_gradients();
+        let patterns = self.patterns();
+        let clip_paths = self.clip_paths();
+        let masks = self.masks();
+        let filters = self.filters();
+        let fontdb = self.fontdb();
+
+        let mut sum = 0;
+        sum += root.size_of(ops);
+        sum += linear_gradients
+            .iter()
+            .map(|x| x.size_of(ops))
+            .sum::<usize>();
+        sum += radial_gradients
+            .iter()
+            .map(|x| x.size_of(ops))
+            .sum::<usize>();
+        sum += patterns.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += clip_paths.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += masks.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += filters.iter().map(|x| x.size_of(ops)).sum::<usize>();
+        sum += fontdb.size_of(ops);
+
+        if ops.has_malloc_enclosing_size_of() {
+            unsafe {
+                sum += ops.malloc_enclosing_size_of(root);
+                if !linear_gradients.is_empty() {
+                    sum += ops.malloc_enclosing_size_of(linear_gradients.as_ptr());
+                }
+                if !radial_gradients.is_empty() {
+                    sum += ops.malloc_enclosing_size_of(radial_gradients.as_ptr());
+                }
+                if !patterns.is_empty() {
+                    sum += ops.malloc_enclosing_size_of(patterns.as_ptr());
+                }
+                if !clip_paths.is_empty() {
+                    sum += ops.malloc_enclosing_size_of(clip_paths.as_ptr());
+                }
+                if !masks.is_empty() {
+                    sum += ops.malloc_enclosing_size_of(masks.as_ptr());
+                }
+                if !filters.is_empty() {
+                    sum += ops.malloc_enclosing_size_of(filters.as_ptr());
+                }
+                sum += ops.malloc_enclosing_size_of(fontdb);
```

Comment:
We don't want to use enclosing_size_of on fontdb, since we already have the owning pointer to it via the `Arc`.